icefoganalytics/sfa-client
# Dev Tooling: Add Basic Test Suite Setup To Project

Fixes https://github.com/icefoganalytics/sfa-client/issues/16

Depends on:
- https://github.com/icefoganalytics/sfa-client/pull/15

# Context

Create a basic test suite setup for the api.
Also created basic migration pipeline. It currently needs _way_ more migrations and some kind of initial schema load, but I'm deferring that for now.

TODO
- [x] initialize enough tables and such to get a basic db test to pass. - _Deferring to future date_

# Notes

1. When using the native knex migrator, you must use snake casing for all table and column names. You must also supply the schema name.
2. When using the db-client you get case conversion, but can't perform schema operations.

# Testing Instructions

1. `dev test` will watch the tests and re-run when a test is saved.
3. Edit `src/api/tests/db/db-client.test.ts` and see that the test suite re-runs
> Note the tests will fail, because the test database doesn't have enough tables or data to run on. This will be handled in future PRs.
